tidy3d/components/data/monitor_data.py

```diff
@@ -1093,6 +1093,10 @@ def to_adjoint_field_sources(self, fwidth: float) -> List[CustomCurrentSource]:
                 if not np.all(values == 0):
                     src_field_components[name] = ScalarFieldDataArray(values, coords=coords)
 
+            # dont include this source if no data
+            if all(fld_cmp is None for _, fld_cmp in src_field_components.items()):
```

Comment:
`src_field_components.values()`?

Reply:
I think pylint used to complain about iterating over values, so I got in the habit of always doing over `items()`. but not really a strong preference.

Reply:
hm I don't see a reason for this. might have been an artifact of possibly unordered dicts, but that's not the case anymore as of python 3.7..